src/Console/Modules/ModulesBaseInstallCommand.php

```diff
@@ -0,0 +1,300 @@
+<?php
+
+namespace Console\Modules;
+
+/*
+ * This file is part of Fork CMS.
+ *
+ * For the full copyright and license information, please view the license
+ * file that was distributed with this source code.
+ */
+
+use Backend\Modules\Extensions\Engine\Model as BackendExtensionsModel;
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+/**
+ * Base module installer\uninstaller class
+ */
+abstract class ModulesBaseInstallCommand extends ContainerAwareCommand
```

Comment:
Abstract classes must be prefixed with Abstract, see rule 2 http://www.php-fig.org/bylaws/psr-naming-conventions/

Reply:
Rename ModuleInstaller/ModuleUninstaller? This will break backward compatibility

Reply:
Deprecate the ModuleInstaller class (and don't make that one abstract), make that an empty class that Extends the AbstractModuleInstaller and change all the installers to extend the AbstractModuleInstaller
That way we have a backwards compatible implementation

In that case you can leave the current exectute method like it is in the ModuleInstaller class and make that an abstract method in the AbstractModuleInstaller class